### PR TITLE
Simplify logic on static designs

### PR DESCRIPTION
**Pull request by calebmkim**
Namely: 

- Better simplification of static guards 
- Set a "cycle limit" for promotion (i.e., if design is above a certain cycle limit we do not promote).

### CODE REVIEW

File: calyx-opt/src/passes/static_promotion.rs [hunk 1]
```diff
@@ -226,12 +226,33 @@ impl StaticPromotion {
...
+                    return Some(split[1]);
+                }
             }
+            None
         });
+        // If they don't give a cycle_limit_str, then we set it to the Default.
+        // If they explicitly say "None" or "none", hten we set it to None.
+        // Makes things a bit confusing.
+        let cycle_limit = if cycle_limit_str.is_none() {
+            Some(33554432)
```
Comment (rachitnigam) [inline calyx-opt/src/passes/static_promotion.rs:243]: How is this number chosen?

Comment (calebmkim) [reply calyx-opt/src/passes/static_promotion.rs:243]: Ah my bad should've explained that. 
I basically looked at what worked well for the LUT numbers for the polybench stuff I've been working on. 33554432=2^25 ended up working well as an upper bound.

Comment (rachitnigam) [reply calyx-opt/src/passes/static_promotion.rs:243]: I think we need to be a little more principled about this perhaps. What happens for higher numbers? We want some argument for why a 25-bit counter being the maximum is a good default choice.

Comment (calebmkim) [reply calyx-opt/src/passes/static_promotion.rs]: Ok finally fixed this to default to none (i.e., no limit). We can always set `cycle-limit` though manually if we want.

**Comment by calebmkim** [comment]
Forgot to merge so doing it now.

**Pull request merged**